libraries/chain/block_log.cpp

```diff
@@ -741,6 +874,51 @@ namespace eosio { namespace chain {
       }));
    }
 
+   bool prune(packed_transaction& ptx) {
+      ptx.prune_all();
+      return true;
+   }
+   
+   void block_log::prune_transactions(uint32_t block_num, const std::vector<transaction_id_type>& ids) {
```

Comment:
I think this should return how many were pruned, so the caller can verify if desired. Also remove the `prune` method above and just add the extra logic to your overloaded lambda.